src/components/CanvasElementsRenderer.tsx

```diff
@@ -64,29 +65,11 @@ export const CanvasElementsRenderer = (props: CanvasElementsRendererProps) => {
         const primitiveIdsWithMouseOver = new Set(
           primitivesHoveredOver.map((p) => p._pcb_drawing_object_id),
         )
-        const newPrimitives = []
-        for (const primitive of primitivesWithoutInteractionMetadata) {
-          const newPrimitive = { ...primitive }
-          if (primitiveIdsWithMouseOver.has(primitive._pcb_drawing_object_id)) {
-            newPrimitive.is_mouse_over = true
-          } else if (
-            primitive._element &&
-            (("pcb_trace_id" in primitive._element &&
-              connectedPrimitiveIds.includes(
-                primitive._element.pcb_trace_id,
-              )) ||
-              ("pcb_port_id" in primitive._element &&
-                connectedPrimitiveIds.includes(
-                  primitive._element.pcb_port_id!,
-                )))
-          ) {
-            newPrimitive.is_in_highlighted_net = true
-          } else {
-            newPrimitive.is_in_highlighted_net = false
-            newPrimitive.is_mouse_over = false
-          }
-          newPrimitives.push(newPrimitive)
-        }
+        const newPrimitives = primitivesToBeHighlighted(
```

Comment:
If a function has more than two variables, dont use positional parameters, used named parameters.

For the name of each parameter of a function, keep the same name as the parent scope, this is called "variable transparency". Variables should not change names between function if they are well named (generally). In this case you renamed promitivesWithoutInteractionMetadata to oldPrimitives

Function names should generally be verbs, not nouns


```suggestion
        const newPrimitives = addInteractionMetadataToPrimitives(
```